src/app/pages/auth/auth-form/auth-form.component.ts

```
@@ -0,0 +1,114 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import {
+  FormGroup,
+  FormControl,
+  Validators,
+  ValidatorFn,
+  AbstractControl,
+  ValidationErrors,
+} from '@angular/forms';
+
+@Component({
+  selector: 'app-auth-form',
+  templateUrl: './auth-form.component.html',
+  styleUrls: ['./auth-form.component.scss'],
+})
+export class AuthFormComponent {
+  @Output() submitBtnFn = new EventEmitter();
+  @Input()
+  set formFor(value: 'Sign In' | 'Sign Up') {
+    this.formType = value;
+    if (this.formType === 'Sign In') {
+      this.authForm.removeControl('passwordRepeat');
+    }
+  }
+
+  checkPasswords: ValidatorFn = (
+    group: AbstractControl
+  ): ValidationErrors | null => {
+    let password = group.get('password')?.value;
+    let passwordRepeat = group.get('passwordRepeat')?.value;
+
+    if (password !== passwordRepeat)
+      group.get('passwordRepeat')?.setErrors({ noRepeat: true });
```

If I remember correctly - ValidatorFn should return an error Obj or null. No need to call setErrors inside. 
Looks like you are using ValidatorFn but it is actually not ValidatorFn